File: src/types/proxy.ts
/**
 * 代理状态
 */
export interface ProxyStatus {
  running: boolean;
  mode: 'rule' | 'global' | 'direct';
  port: number;
  socks_port: number;
  mixed_port?: number;
  system_proxy: boolean;
  enhanced_mode?: boolean;
  allow_lan?: boolean;
  ipv6?: boolean;
  tcp_concurrent?: boolean;
}

/**
 * 代理服务器信息
 */
export interface ProxyServerInfo {
  name: string;
  type: string;
  server: string;
  port: number;
  udp?: boolean;
}

/**
 * 代理组
 */
export interface ProxyGroup {
  name: string;
  type: string;
  now?: string;
  all: string[];
}

/**
 * 代理节点
 */
export interface ProxyNode {
  name: string;
  type: string;
  delay?: number;
  udp: boolean;
  selected: boolean;
}

/**
 * 流量数据
 */
export interface TrafficData {
  up: number;
  down: number;
}

/**
 * 连接信息
 */
export interface Connection {
  id: string;
  metadata: ConnectionMetadata;
  upload: number;
  download: number;
  start: string;
  chains: string[];
  rule: string;
  rulePayload: string;
}

/**
 * 连接元数据
 */
export interface ConnectionMetadata {
  network: string;
  type: string;
  sourceIP: string;
  destinationIP: string;
  sourcePort: string;
  destinationPort: string;
  host: string;
  dnsMode: string;
  process?: string;
  processPath?: string;
}

/**
 * 连接列表响应
 */
export interface ConnectionsResponse {
  connections: Connection[];
  downloadTotal: number;
  uploadTotal: number;
}

/**
 * 代理模式类型
 */
export type ProxyMode = 'rule' | 'global' | 'direct';

/**
 * 代理模式配置
 */
export const PROXY_MODES: { value: ProxyMode; label: string; description: string }[] = [
  { value: 'rule', label: '规则模式', description: '根据规则自动分流' },
  { value: 'global', label: '全局模式', description: '所有流量走代理' },
  { value: 'direct', label: '直连模式', description: '所有流量直连' },
];

/**
 * 规则项（来自 mihomo API）
 */
export interface RuleItem {
  type: string;
  payload: string;
  proxy: string;
}

/**
 * 规则列表响应
 */
export interface RulesResponse {
  rules: RuleItem[];
}

/**
 * 核心版本信息
 */
export interface VersionInfo {
  version: string;
  meta: boolean;
}

File: src-tauri/src/models/proxy.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

/// 代理状态
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ProxyStatus {
    pub running: bool,
    pub mode: String,
    pub port: u16,
    pub socks_port: u16,
    pub mixed_port: Option<u16>,
    pub system_proxy: bool,
    pub allow_lan: bool,
    pub ipv6: bool,
    pub tcp_concurrent: bool,
}

impl Default for ProxyStatus {
    fn default() -> Self {
        Self {
            running: false,
            mode: "rule".to_string(),
            port: 7890,
            socks_port: 7891,
            mixed_port: None,
            system_proxy: false,
            allow_lan: false,
            ipv6: false,
            tcp_concurrent: false,
        }
    }
}

/// 代理节点信息
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ProxyInfo {
    pub name: String,
    #[serde(rename = "type")]
    pub proxy_type: String,
    #[serde(default)]
    pub udp: bool,
    #[serde(default)]
    pub history: Vec<DelayHistory>,
    #[serde(default)]
    pub all: Vec<String>,
    #[serde(default)]
    pub now: Option<String>,
}

/// 延迟历史记录
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DelayHistory {
    pub time: String,
    pub delay: u32,
}

/// 代理列表响应
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ProxiesResponse {
    pub proxies: HashMap<String, ProxyInfo>,
}

/// 延迟测试响应
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DelayResponse {
    pub delay: u32,
}

/// 连接信息
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Connection {
    pub id: String,
    pub metadata: ConnectionMetadata,
    pub upload: u64,
    pub download: u64,
    pub start: String,
    pub chains: Vec<String>,
    pub rule: String,
    #[serde(rename = "rulePayload")]
    pub rule_payload: String,
}

/// 连接元数据
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ConnectionMetadata {
    pub network: String,
    #[serde(rename = "type")]
    pub conn_type: String,
    #[serde(rename = "sourceIP")]
    pub source_ip: String,
    #[serde(rename = "destinationIP")]
    pub destination_ip: String,
    #[serde(rename = "sourcePort")]
    pub source_port: String,
    #[serde(rename = "destinationPort")]
    pub destination_port: String,
    pub host: String,
    #[serde(rename = "dnsMode")]
    pub dns_mode: String,
    pub process: Option<String>,
    #[serde(rename = "processPath")]
    pub process_path: Option<String>,
}

/// 连接列表响应
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ConnectionsResponse {
    #[serde(default)]
    pub connections: Vec<Connection>,
    #[serde(rename = "downloadTotal")]
    pub download_total: u64,
    #[serde(rename = "uploadTotal")]
    pub upload_total: u64,
}

/// 流量数据
#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub struct TrafficData {
    pub up: u64,
    pub down: u64,
}

/// 版本信息
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct VersionInfo {
    pub version: String,
    #[serde(default)]
    pub meta: bool,
}

/// 前端代理节点显示
#[allow(dead_code)]
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ProxyNode {
    pub name: String,
    #[serde(rename = "type")]
    pub node_type: String,
    pub delay: Option<u32>,
    pub udp: bool,
    pub selected: bool,
}

/// 代理组
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ProxyGroup {
    pub name: String,
    #[serde(rename = "type")]
    pub group_type: String,
    pub now: Option<String>,
    pub all: Vec<String>,
}

/// 规则项
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RuleItem {
    #[serde(rename = "type")]
    pub rule_type: String,
    pub payload: String,
    pub proxy: String,
}

/// 规则列表响应
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct RulesResponse {
    pub rules: Vec<RuleItem>,
}
